common/src/main/java/com/microsoft/identity/common/internal/controllers/ExceptionAdapter.java

```diff
@@ -188,24 +198,143 @@ public static ServiceException getExceptionFromTokenErrorResponse(@NonNull final
+     */
+
+    private static void setHttpResponseUsingTokenErrorResponse(@NonNull final ServiceException exception,
+            @NonNull final TokenErrorResponse errorResponse){
+
+        try {
+            exception.setHttpResponse(
+                    synthesizeHttpResponse(
+                    errorResponse.getStatusCode(),
+                    errorResponse.getResponseHeadersJson(),
+                    errorResponse.getResponseBody()));
+        } catch (JSONException e) {
             Logger.warn(
-                    TAG + methodName,
+                    TAG,
                     "Failed to deserialize error data: status, headers, response body."
             );
         }
 
-        return outErr;
+    }
+
+    /**
+     * Helper method to get uid from home account id
+     * V2 home account format : <uid>.<utid>
+     * V1 : it's stored as <uid>
+     *
+     * @param homeAccountId
+     * @return valid uid or null if it's not in either of the format.
+     */
+    @Nullable
+    public static String getUIdFromHomeAccountId(@Nullable String homeAccountId) {
+        //TODO: This method is from BrokerOperationParameterUtils...
+        // seems like this is not broker specific per se and should move to somewhere better
+        final String methodName = ":getUIdFromHomeAccountId";
+        final String DELIMITER_TENANTED_USER_ID = ".";
+        final int EXPECTED_ARGS_LEN = 2;
+        final int INDEX_USER_ID = 0;
+
+        if (!TextUtils.isEmpty(homeAccountId)) {
+            final String[] homeAccountIdSplit = homeAccountId.split(
+                    Pattern.quote(DELIMITER_TENANTED_USER_ID)
+            );
+
+            if (homeAccountIdSplit.length == EXPECTED_ARGS_LEN) {
+                com.microsoft.identity.common.internal.logging.Logger.info(TAG + methodName,
+                        "Home account id is tenanted, returning uid "
+                );
+                return homeAccountIdSplit[INDEX_USER_ID];
+            } else if (homeAccountIdSplit.length == 1) {
+                com.microsoft.identity.common.internal.logging.Logger.info(TAG + methodName,
+                        "Home account id not tenanted, it's the uid added by v1 broker "
+                );
+                return homeAccountIdSplit[INDEX_USER_ID];
+            }
+        }
+
+        com.microsoft.identity.common.internal.logging.Logger.warn(TAG + methodName,
+                "Home Account id doesn't have uid or tenant id information, returning null "
+        );
+
+        return null;
+    }
+
+    private static void setIntuneExceptionProperties(
+            @NonNull final IntuneAppProtectionPolicyRequiredException exception,
+            @NonNull final BrokerSilentTokenCommandParameters requestParameters) {
+
+        com.microsoft.identity.common.internal.logging.Logger.info(TAG, "Setting properties to IntuneAppProtectionPolicyRequiredException ");
```

Comment:
nit: remove extra `com.microsoft.identity.common.internal.logging` import. Here and below.

Reply:
Done.